test/e2e/apps/cronjob.go

```diff
@@ -412,17 +434,32 @@ var _ = SIGDescribe("CronJob", func() {
 		framework.ExpectEqual(string(createdCronJob.UID), statusUID, fmt.Sprintf("createdCronJob.UID: %v expected to match statusUID: %v ", createdCronJob.UID, statusUID))
 
 		// CronJob resource delete operations
-		ginkgo.By("deleting a collection")
-		expectFinalizer := func(cj *batchv1beta1.CronJob, msg string) {
+		expectFinalizer := func(cj *batchv1.CronJob, msg string) {
 			framework.ExpectNotEqual(cj.DeletionTimestamp, nil, fmt.Sprintf("expected deletionTimestamp, got nil on step: %q, cronjob: %+v", msg, cj))
 			framework.ExpectEqual(len(cj.Finalizers) > 0, true, fmt.Sprintf("expected finalizers on cronjob, got none on step: %q, cronjob: %+v", msg, cj))
 		}
+
+		ginkgo.By("deleting")
+		cjTemplate.Name = "for-removal"
+		forRemovalCronJob, err := cjClient.Create(context.TODO(), cjTemplate, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		err = cjClient.Delete(context.TODO(), forRemovalCronJob.Name, metav1.DeleteOptions{})
+		framework.ExpectNoError(err)
+		cj, err := cjClient.Get(context.TODO(), forRemovalCronJob.Name, metav1.GetOptions{})
+		// If controller does not support finalizers, we expect a 404.  Otherwise we validate finalizer behavior.
+		if err == nil {
+			expectFinalizer(cj, "deleting cronjob")
+		} else {
+			framework.ExpectEqual(apierrors.IsNotFound(err), true, fmt.Sprintf("expected 404, got %v", err))
+		}
+
+		ginkgo.By("deleting a collection")
 		err = cjClient.DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: "special-label=" + f.UniqueName})
 		framework.ExpectNoError(err)
 		cjs, err = cjClient.List(context.TODO(), metav1.ListOptions{LabelSelector: "special-label=" + f.UniqueName})
 		framework.ExpectNoError(err)
-		// Should have <= 1 items since some cronjobs might not have been deleted yet due to finalizers
-		framework.ExpectEqual(len(cjs.Items) <= 1, true, "filtered list should be <= 1")
+		// Should have <= 2 items since some cronjobs might not have been deleted yet due to finalizers
+		framework.ExpectEqual(len(cjs.Items) <= 2, true, "filtered list should be <= 2")
```

Comment:
why the change from 1 to 2?

Reply:
This PR promotes the test to conformance, but does not include the change to make the controller use the v1 API?

This test passed in the ga-only CI job, which turns off beta APIs. If the controller is still using v1beta1 in this branch, does the fact that the test passed mean this test does not check anything the controller does? Wouldn't we expect to verify basic functionality of cronjob from a user's perspective?

Reply:
> why the change from 1 to 2?

This test previously was missing single delete, there was only delete collection. I've added that one bit to have a full API operation coverage. This means that during check there might be 2 cronjobs, so I'm checking for up to 2 cronjobs marked for deletion. One from a single delete, the other from delete collection. 

> This test passed in the ga-only CI job, which turns off beta APIs. If the controller is still using v1beta1 in this branch, does the fact that the test passed mean this test does not check anything the controller does? Wouldn't we expect to verify basic functionality of cronjob from a user's perspective?

lemme rebase this on top of both api and controller changes to ensure this works as expected.

Reply:
Also I don't expect this to fail in any way, aside from my personal testing, all of these tests (with the exception of that last change) are part of a regular e2e (non-conformance) for quite a while now and didn't had issues at least in the past weeks.

Reply:
> This test previously was missing single delete, there was only delete collection. I've added that one bit to have a full API operation coverage.

ah, got it